File: design-patterns/template-method/plain_text_report.rb
require_relative 'report'

class PlainTextReport < Report
  def output_start
  end
  
  def output_head
    p "**** #{title} ****"
    puts
  end
  
  def output_body_start
  end
  
  def output_line(line)
    p line
  end
  
  def output_body_end
  end
  
  def output_end
  end
end

File: design-patterns/cs/Patrones/Patrones/TemplateMethod/PlainTextReport.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Patrones.TemplateMethod
{
    public class PlainTextReport : Report
    {
        protected override string OutputStart()
        {
            return string.Empty;
        }

        protected override string OutputHead()
        {
            return "*****" + Title + "*****" + Environment.NewLine + Environment.NewLine;
        }

        protected override string OutputBodyStart()
        {
            return string.Empty;
        }

        protected override string OutputBodyEnd()
        {
            return string.Empty;
        }

        protected override string OutputEnd()
        {
            return string.Empty;
        }

        protected override string OutputLine(string line)
        {
            return line + Environment.NewLine;
        }
    }
}
